### PR TITLE
Fetch file

### PR DESCRIPTION
**Pull request by lastmjs**
This PR implements `file://` for fetch from a canister, not from the frontend. It also has example code and tests. This is a very simple version of file://

### CODE REVIEW

**Comment by bdemann** [comment]
# What is the feature?

## Does it have an issue?
No. Looks like its a bullet point of https://github.com/demergent-labs/azle/issues/1691

## What is it doing?
### Explanation
Implements fetch file presumably for the frontend since
https://github.com/demergent-labs/azle/pull/1706 is going to take care of the
backend

### Questions
- What do I need to know about the fetch file protocol before I can competently
preform this review?

### Answers?
#### What is the File protocol?
The file protocol is used to get files on the local file system[^cn]. So I would
imagine that on the client side the file protocol would get files on the user's
or developer's computer and on the canister side it would be used to get files
in the canister's file system. So I would expect that on the client side there
isn't much to do besides just pass off to the default implementation of fetch
right?

[^cn]: Citation Needed

#### Considerations
##### Security restrictions
According to Gemini "Most browsers and environments restrict file protocol
fetching for security reasons. Fetching from arbitrary locations on the user's
device could be a privacy or security risk. Your implementation might need to
consider restrictions or permissions." So in the event that we can't just pass
it off we will have to consider that.

##### Edge Cases
##### Non-existent Files
We might have to handle non-existent files gracefully if we can't just use the 
default fetch
##### Permissions
We might have to worry about permissions if we can't just use the default 
fetch

In short I hope we can just use the default fetch.


# Expectations
## Tests

I guess I would expect a front end that is able to get a local file and... do
something with it? Send it to the canister?

## Code

Hopefully just hand off to the default implementation of fetch.

**Comment by bdemann** [comment]
Okay this is definitely a canister side fetch, that's a horse of a different color. Let's start over

**Comment by bdemann** [comment]
## Should this be implemented?
I've started reading some stuff about how node fetch doesn't actually implement the file protocol, so we might not should do it either. I'll do some more research including working more with tensor flow to see what exactly it needs the file protocol for and make sure we are understanding its needs correctly.

**Review by bdemann — APPROVED** [review]

File: src/lib/fetch.ts [hunk 1]
```diff
@@ -66,12 +67,31 @@ export async function azleFetch(input: any, init?: any): Promise<any> {
...
         };
-    } else {
-        // TODO http request to the management canister
+    }
+
+    if (url.protocol === 'file:') {
+        const path = `${url.hostname}${url.pathname}`;
+        const contents = await readFile(path);
+
+        return {
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/lib/fetch.ts:83]: Response object is not part of QuickJS so we have to make our own, that is outside of the scope of this PR and is currently a WIP feature.
